nvflare/fuel/f3/streaming/stream_utils.py

```diff
@@ -22,11 +23,33 @@
 STREAM_THREAD_POOL_SIZE = 128
 ONE_MB = 1024 * 1024
 
-stream_thread_pool = ThreadPoolExecutor(STREAM_THREAD_POOL_SIZE, "stm")
 lock = threading.Lock()
 sid_base = int((time.time() + os.getpid()) * 1000000)  # microseconds
 stream_count = 0
 
+log = logging.getLogger(__name__)
+
+
+class CheckedExecutor(ThreadPoolExecutor):
+    """This executor ignores task after shutting down"""
+
+    def __init__(self, max_workers=None, thread_name_prefix=""):
+        super().__init__(max_workers, thread_name_prefix)
+        self.stopped = False
+
+    def shutdown(self, wait=True):
+        super().shutdown(wait)
+        self.stopped = True
```

Comment:
Should we move self.stopped before the super().shutdown(wait)?
I guess it comes down to if someone call "shutdown", do we block submit right away?
Or we wait for all things to come back then block.

Reply:
Good idea. Changed.